marco/AutamationPractice/src/main/java/at05ui/core/utils/JsonReader.java

```diff
@@ -0,0 +1,145 @@
+     */
+    private JSONObject getJSONObject(String key) {
+        return (JSONObject) jsonObjectMain.get(key);
+    }
+
+    /**
+     * Gets the jsonObject value given the key from a jsonObject
+     */
+    private JSONObject getJSONObject(JSONObject jsonObject, String key) {
+        return (JSONObject) jsonObject.get(key);
+    }
+
+    /**
+     * Gets the string value given the key from an specific jsonObject
+     */
+    private String getKeyValueFromJSONObject(JSONObject jsonObject, String key) {
+        return (String) jsonObject.get(key);
+    }
+
+    /**
+     * Gets the jsonObject which key and value matches the given parameters from a jsonArray
+     */
+    private JSONObject getJSONObjectFromArrayById(String objectName, String idKey, String idValue) {
+        JSONObject jsonObject = null;
+        JSONArray arr = (JSONArray) jsonObjectMain.get(objectName);
+        for (Object anArr : arr) {
+            jsonObject = (JSONObject) anArr;
+            if (jsonObject.get(idKey).equals(idValue)) {
+                break;
+            }
+        }
+        return jsonObject;
+    }
+
+    /**
+     * Gets a jsonObject from the main json and then gets a value given the key
+     */
+    public String getKeyValue(String objectName, String key) {
+        JSONObject jsonObject = getJSONObject(objectName);
+        return getKeyValueFromJSONObject(jsonObject, key);
+    }
+
+    /**
+     * Gets the jsonObject from the jsonArray and then gets a value given the key
+     */
+    public String getKeyValue(String objectName, String idKey, String idValue, String key) {
+        JSONObject jsonObject = getJSONObjectFromArrayById(objectName, idKey, idValue);
+        return getKeyValueFromJSONObject(jsonObject, key);
+    }
+
+    /**
+     * Gets the jsonObject from the jsonArray and then gets a jsonObject value and then gets a value
+     * given the key
+     */
+    public String getKeyValue(String objectName, String idKey, String idValue, String objectKey,
+        String key) {
+        JSONObject jsonObject = getJSONObjectFromArrayById(objectName, idKey, idValue);
+        JSONObject jsonObjectInternal = getJSONObject(jsonObject, objectKey);
+        return getKeyValueFromJSONObject(jsonObjectInternal, key);
+    }
+
+    /**
+     * Gets a hashMap which key is the user type and value is a hashMap that contains the values for
+     * Name, Password
+     */
+    public HashMap<String, Object> getArrayKeyValues(String objectName, String idKey, String idValue,
+        String key) {
+        JSONObject jsonObject = getJSONObjectFromArrayById(objectName, idKey, idValue);
+        JSONArray arr = (JSONArray) jsonObject.get(key);
+
+        HashMap<String, Object> arrMap = new HashMap<>();
+
+        //Iterate all the array of jsons to add an entry (key, map) in arrValues by each json
+        for (Object anArr : arr) {
+            JSONObject jsonArrElement = (JSONObject) anArr;
+
```

Comment:
Remove unnecessary spaces, you could use Ctrl+Alt+L

Reply:
I am using the check style of google in my project, but I have not implemented in Silvia's files.
Thanks a lot for the observation.